test/fail_compilation/issue22820.d

```diff
@@ -0,0 +1,45 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/issue22820.d(126): Error: need upper and lower bound to slice pointer, perhaps you meant `(*s1)[]`
+fail_compilation/issue22820.d(127): Error: need upper and lower bound to slice pointer, perhaps you meant `(*s2)[]`
+fail_compilation/issue22820.d(128): Error: need upper and lower bound to slice pointer, perhaps you meant `(*s3)[]`
+---
+*/
+
+#line 100
+
+// normal functions
+struct S1 {
+	int[] opIndex() { return a; }
+	int[] a;
+}
+
+// opSlice alternative
+struct S2 {
+	int[] opSlice() { return a; }
```

Comment:
AFAIK, the specification only mentions opIndex, but it works with opSlice too. Can someone clarify this to me? This seems a case where users can implement both and the compiler doesn't warn you about it. Perhaps add a subsequent error or prohibit one of them

Reply:
https://dlang.org/spec/operatoroverloading.html#slice

Point 9, `opSlice` is a backward-compatible shenanigans.

Reply:
Ah right, missed that last one. I know this is not related to this PR at all but is there any plan to deprecate this?

Reply:
Not that I know of. And IDK if deprecation would be of much value for the user (I actually thought `opSlice` was the right way to do it before looking it up).